File: src/main/kotlin/com/feakin/intellij/FeakinFileType.java
package com.feakin.intellij;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class FeakinFileType extends LanguageFileType {
    public static final FeakinFileType INSTANCE = new FeakinFileType();

    private FeakinFileType() {
        super(FeakinLanguage.INSTANCE);
    }

    @Override
    public @NotNull String getName() {
        return "Charj File";
    }

    @Override
    public @NotNull String getDescription() {
        return "Charj language file";
    }

    @Override
    public @NotNull String getDefaultExtension() {
        return "cj";
    }

    @Override
    public @Nullable Icon getIcon() {
        return FeakinIcons.FILE;
    }
}

File: src/main/kotlin/com/feakin/intellij/FeakinFileType.kt
package com.feakin.intellij

import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.vfs.VirtualFile
import javax.swing.Icon

object FeakinFileType : LanguageFileType(FeakinLanguage) {

    override fun getName(): String = "Feakin File"

    override fun getIcon(): Icon = FkIcons.FILE

    override fun getDefaultExtension(): String = "fkl"

    override fun getCharset(file: VirtualFile, content: ByteArray): String = "UTF-8"

    override fun getDescription(): String = "Feakin DSL file"
}